lib/src/http.rs

```diff
@@ -447,9 +448,26 @@ impl L7ListenerHandler for HttpListener {
         */
         let host = unsafe { from_utf8_unchecked(hostname) };
 
-        self.fronts
+        let res = self
+            .fronts
             .lookup(host.as_bytes(), uri.as_bytes(), method)
-            .ok_or(FrontendFromRequestError::NoClusterFound)
+            .ok_or(FrontendFromRequestError::NoClusterFound);
+
+        let now = Instant::now();
+
+        if let Ok(r) = res.as_ref() {
+            if let Route::ClusterId(c) = r {
+                time!(
+                    "frontend_matching_time",
+                    &c,
+                    (now - start).whole_milliseconds()
+                );
+            }
+        } else {
+            incr!("http.failed_backend_matching")
+        }
+
+        res
```

Comment:
please rename `res` to `result` (if present at all), `r` to `route`, and `c` to `cluster`.